src/Body.php

```diff
@@ -0,0 +1,33 @@
+<?php
+
+namespace Http\Client;
+
+use Psr\Http\Message\StreamInterface;
+
+/**
+ * Allows special input as body
+ *
+ * @author Márk Sági-Kazár <mark.sagikazar@gmail.com>
+ */
+interface Body
+{
+    /**
+     * Returns a set of headers which is needed to correctly send the body
+     *
+     * Note: these headers get overwritten by headers manually passed to the client
+     *
+     * Content-Length is calculated automatically if possible
+     *
+     * @return array
+     */
+    public function getContentHeaders();
+
+    /**
+     * Convert data to a format which can be used to create a proper PSR-7 Stream
+     *
+     * @return string|StreamInterface
```

Comment:
nonetheless we should explain what the string we allow to return here is. if we allow a string at all - it seems like an ambiguity. ideally we would even eliminate string as return completely. otherwise a consumer needs to check whether it got a string or a stream and do different things...